aepsych/strategy.py

```diff
@@ -328,10 +335,12 @@ def add_data(
         self.x, self.y, self.n = self.normalize_inputs(x, y)
         self._model_is_fresh = False
 
-    def fit(self):
+    def fit(self) -> None:
         if self.can_fit:
             if self.keep_most_recent is not None:
                 try:
+                    assert self.model is not None, "a model is needed here!"
```

Comment:
Logically can_fit should be enough, but I still get the following error if I remove the assertion:
 `error: Item "None" of "AEPsychMixin | None" has no attribute "fit"  [union-attr] self.model.fit( `

Reply:
Assert probably due to above Optional on can_fit. Wherever assert ends up being weirdly needed, it's likely because of an odd typing somewhere else that should be fixed instead.